workbench-core/accounts/src/services/projectService.ts

```diff
@@ -56,20 +62,143 @@ export default class ProjectService {
     }
   }
 
+  // TODO--delete after dynamic Authz
+  private _mockGetUserGroups(): string[] {
+    return [];
+  }
+
   /**
    * List projects
    *
+   * @param request - the optional fields to list projects
+   * @param user - authenticated user listing the projects
    * @returns Project entries in DDB
    */
-  public async listProjects(): Promise<{ data: Project[] }> {
-    const queryParams = {
+  // TODO--add filter support
+  public async listProjects(
+    user: AuthenticatedUser,
+    request?: ListProjectsRequest
+  ): Promise<{ data: Project[]; paginationToken: string | undefined }> {
+    const pageSize = request ? request.pageSize : undefined;
+    let paginationToken = request ? request.paginationToken : undefined;
+    // Get user groups--TODO implement after dynamic AuthZ
+    // const userGroupsForCurrentUser: string[] = await this._dynamicAuthorizationService.getUserGroups(user.id);
+    const userGroupsForCurrentUser: string[] = this._mockGetUserGroups(); // mock so the tests work
+
+    // If no group membership, return
+    if (_.isEmpty(userGroupsForCurrentUser)) {
+      return { data: [], paginationToken: undefined };
+    }
+
+    // If IT Admin, return all with pagination
+    if (userGroupsForCurrentUser.length === 1) {
+      if (userGroupsForCurrentUser[0] === 'ITAdmin') {
+        let queryParams: QueryParams = {
+          key: { name: 'resourceType', value: this._resourceType },
+          index: 'getResourceByCreatedAt',
+          limit: pageSize && pageSize >= 0 ? pageSize : DEFAULT_API_PAGE_SIZE
+        };
+
+        queryParams = addPaginationToken(paginationToken, queryParams);
+
+        const projectsResponse = await this._aws.helpers.ddb.query(queryParams).execute();
+
+        paginationToken = getPaginationToken(projectsResponse);
+
+        return Promise.resolve({
+          data: projectsResponse.Items as unknown as Project[],
+          paginationToken: paginationToken
+        });
+      }
+      // If member of 1 group, get project item
+      if (userGroupsForCurrentUser[0] !== 'ITAdmin') {
+        const projectId = userGroupsForCurrentUser[0].split('#')[0];
+        const project = await this.getProject(projectId);
+        return Promise.resolve({ data: [project], paginationToken: undefined });
+      }
+    }
+
+    // Else, member of more than 1 group, query for project items with pagination
+    // welcome to party town
+    const projectIds: string[] = userGroupsForCurrentUser.map(
+      (projectGroup: string) => projectGroup.split('#')[0]
+    );
+
+    // i = 0
+    let filterExpression = 'id = :proj1';
+    const expressionAttributeValues: { [key: string]: string } = { ':proj1': projectIds[0] };
+    for (let i = 1; i < projectIds.length; i++) {
+      // build filter string "id = :proj1 OR id = :proj2 OR ... OR id = projN+1"
```

Comment:
Was `IN` operator considered?
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.OperatorsAndFunctions.html

Reply:
This piece of code (getting multiple projects) will be rewritten with the new design as mentioned in meetings today so I am going to resolve this convo.